### DISCUSSION TITLE
publish  \vendor\filament\forms\src\Components\CheckboxList.php

### DISCUSSION THREAD

**Discussion by juangdnar** [filamentphp/filament#9567] [Help] [answered]
It is possible to publish one of the \vendor\filament\forms\src\Components files outside the vendor folder to customize it, specifically I need to make changes to CheckboxList.php but I require that it not be overwritten when the filament version is updated.

thanks in advanced.

**Comment by leandrocfe** [comment, marked as answer]
Create a custom class in your app
```php
//app/Filament/Forms/Components/CheckboxList.php
namespace App\Filament\Forms\Components;

use Filament\Forms\Components\CheckboxList as BaseCheckboxList;

class CheckboxList extends BaseCheckboxList
{
    protected string $view = 'filament.forms.components.checkbox-list';
}
```

Create a custom view in `resources/views/filament/forms/components/checkbox-list.blade.php`

Copy the original content from the vendor and edit what you want.

Use in your form 
```php
\App\Filament\Forms\CheckboxList::make('field')
```

**Reply by jcgmalave** [reply]
it works!!! :)
